[12.0][ADD] Allow to import orders with a certain delay

StefanRijnhart opened the pull request:
This is useful when orders are imported frequently as it prevents orders
from being found unpaid if the transfer to status 'paid' is delayed at the
payment provider (set this value to the duration of that delay).

hbrunn commented on connector_magento/models/magento_storeview/common.py:
```diff
@@ -74,32 +74,35 @@ def import_sale_orders(self):
-                'from_date': from_date,
-                'to_date': import_start_time,
+                'from_date': (fields.Datetime.to_string(from_date)
+                              if from_date else None),
```
if `False` is also fine, you don't need this check: https://github.com/OCA/OCB/blob/12.0/odoo/fields.py#L1790

StefanRijnhart replied:
Nooooo. I had troubles reproducing this, but I'll try again now.

StefanRijnhart replied:
I don't understand my initial answer anymore, but the answer is that False is not a valid value for this key.

hbrunn reviewed (changes requested):
I think it would be good to explain in the readme when to use the delay, and when to rely on `IMPORT_DELTA_BUFFER` - my assumption is you'll want the delay on a busy site?

StefanRijnhart committed:
[ADD] Allow to import orders with a certain delay

This is useful when orders are imported frequently as it prevents orders
from being found unpaid if the transfer to status 'paid' is delayed at the
payment provider (set this value to the duration of that delay).

StefanRijnhart commented:
@hbrunn sorry for losing sight of this PR! I've documented this parameter now in the RST docs where I'm not mentioning `IMPORT_DELTA_BUFFER` as this is a technical and somewhat trivial qualification of the concept of 'now'.

hbrunn committed:
[FIX] connector_magento: Use datetime objects in filters, not strings

StefanRijnhart commented:
@hbrunn thanks for the fix!

dreispt commented:
/ocabot merge nobump
